lazer/sdk/rust/protocol/src/router.rs

```diff
@@ -77,13 +74,20 @@ impl Price {
     pub fn into_inner(self) -> NonZeroI64 {
         self.0
     }
-}
 
-impl TryInto<f64> for Price {
-    type Error = anyhow::Error;
+    pub fn to_f64(self, exponent: u32) -> anyhow::Result<f64> {
+        Ok(self.0.get() as f64 / 10i64.checked_pow(exponent).context("overflow")? as f64)
+    }
+
+    pub fn mul(self, rhs: Price, rhs_exponent: u32) -> anyhow::Result<Price> {
+        let left_value = i128::from(self.0.get());
+        let right_value = i128::from(rhs.0.get());
 
-    fn try_into(self) -> Result<f64, Self::Error> {
-        Ok(self.0.get() as f64 / 10i64.checked_pow(Self::TMP_EXPONENT).context("overflow")? as f64)
+        let value = left_value * right_value / 10i128.pow(rhs_exponent);
+        let value = value.try_into()?;
+        NonZeroI64::new(value)
+            .context("zero price is unsupported")
+            .map(Self)
```

Comment:
any reason you removed the traits?

Reply:
With the need to pass an exponent to the `mul` and `f64 cast` functionalities, the traits are no longer suitable. So I it's has been changed to functions.